src/foam/nanos/ruler/Rule.js

```diff
@@ -230,6 +234,48 @@
           ruler.getDelegate().cmd_(x.put("OBJ", obj), getCmd());
         }
       `
+    },
+    {
+      name: 'reverseAction',
+      args: [
+        {
+          name: 'obj',
+          type: 'foam.core.FObject'
+        },
+        {
+          name: 'x',
+          type: 'foam.core.X'
+        }
+      ],
+      documentation: 'if one of the rules in a group throws an exception we need a way to reverse actions for previously executed rules.',
```

Comment:
What are examples of when you would need this?

Reply:
Alternative to canExecute method. Example: If we have transaction limit for sender passed fine and updated and then receiver's limit check fails, we want to revert change on sender limit.

Reply:
If we executed the action after all actions passed, wouldn't that be simpler? So if an action had a before action and an after action rather than a before action and then a reverse action?

Reply:
Our rules execution is not synchronized, so even if canExecute method returns true at the moment of execution, there is a possibility that by the time actual action will be applied, it would return false. Also logic has to be duplicated for rules whose action directly depend on the check.